### ISSUE TITLE
Problems in Virconv-L experiments

### ISSUE THREAD

**Issue by weilli** [hailanyi/VirConv#21]
This work is excellent.
I just run the experiments in Virconv-L and get these seemingly abnormal results on validation set, which are much higher than those in the paper (0.7.0.5,0.5):

2023-04-28 09:17:56,552   INFO  ==> Checkpoint trained from version: pcdet+0.3.0+0000000
2023-04-28 09:17:56,590   INFO  ==> Done (loaded 467/467)
2023-04-28 09:17:56,596   INFO  *************** EPOCH 50 EVALUATION *****************
2023-04-28 09:19:24,048   INFO  *************** Performance of EPOCH 50 *****************
2023-04-28 09:19:24,049   INFO  Generate label finished(sec_per_example: 0.0232 second).
2023-04-28 09:19:24,049   INFO  recall_roi_0.3: 0.955199
2023-04-28 09:19:24,049   INFO  recall_rcnn_0.3: 0.955199
2023-04-28 09:19:24,049   INFO  recall_roi_0.5: 0.949582
2023-04-28 09:19:24,049   INFO  recall_rcnn_0.5: 0.949582
2023-04-28 09:19:24,049   INFO  recall_roi_0.7: 0.865324
2023-04-28 09:19:24,049   INFO  recall_rcnn_0.7: 0.865324
2023-04-28 09:19:24,050   INFO  Average predicted number of objects(3769 samples): 4.964
2023-04-28 09:19:28,525   INFO  Car AP@0.70, 0.70, 0.70:
bbox AP:97.7144, 96.5966, 89.5965
bev  AP:90.3365, 88.6169, 88.1132
3d   AP:89.5594, 86.2943, 85.5232
aos  AP:97.70, 96.48, 89.49
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:99.0701, 97.6235, 95.2357
bev  AP:95.9682, 91.8133, 91.3687
3d   AP:92.8031, 87.8624, 85.5142
aos  AP:99.06, 97.51, 95.08
Car AP@0.70, 0.50, 0.50:
bbox AP:97.7144, 96.5966, 89.5965
bev  AP:97.7790, 96.5670, 89.5987
3d   AP:97.7242, 96.4896, 89.5544
aos  AP:97.70, 96.48, 89.49
Car AP_R40@0.70, 0.50, 0.50:
bbox AP:99.0701, 97.6235, 95.2357
bev  AP:99.0634, 97.7069, 95.2778
3d   AP:99.0315, 97.6308, 95.1955
aos  AP:99.06, 97.51, 95.08

Could you tell me where I should look for the problem? I have checked the split of the training set and the verification set and there are no problems.

**Comment by hailanyi** [comment]
1. Please align the performance in readme, as the performance is somewhat related to runing environment, please ensure that you run the code with the same environment as in readme.  
2. In most cases, VirConv-L can get an AP of 88+. If you cannot obtain a similar performance, please try multiple times. The results in readme and paper is the best one after multiple training.

**Comment by weilli** [comment]
> 

However, the results (97.7242, 96.4896, 89.5544) are much higher than the paper (93.36 88.71 85.83).

**Comment by hailanyi** [comment]
Here, your AP R40 (IoU=0.7) results are （92.8031, 87.8624, 85.5142）instead of （97.7242, 96.4896, 89.5544）. The  （97.7242, 96.4896, 89.5544）are AP R40 (IoU=0.5) ， which not reported in paper.

**Comment by weilli** [comment]
Thanks for your reply. I figured it out.

**Comment by Omar-Aldade** [comment]
I ran the test script on the pretrained VirConv_L2 Model multiple times on the same machine and got slightly different results each time. Is there a random element to the model, such as  random sampling?   Thank you again for your constant support

**Comment by hailanyi** [comment]
Yes, the random element exists. Since multiple 3D voxels maybe projected to a single 2D pixel, the NRConv only randomly keeps single 3D feature for 2D sparse conv. So the results are not consistant.
